packages/app/src/cli/utilities/app-command.ts

```diff
@@ -1,8 +1,24 @@
 import {configurationFileNames} from '../constants.js'
-import Command from '@shopify/cli-kit/node/base-command'
+import {AppInterface} from '../models/app/app.js'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import BaseCommand from '@shopify/cli-kit/node/base-command'
 
-export default abstract class AppCommand extends Command {
+/**
+ * By forcing all commands to return `AppCommandOutput` we can be sure that during the run of each command we:
+ * - Authenticate the user (PENDING)
+ * - Load an app
+ */
+export interface AppCommandOutput {
+  // session: PartnersSession (PENDING)
+  app: AppInterface
+}
+
+export default abstract class AppCommand extends BaseCommand {
   environmentsFilename(): string {
     return configurationFileNames.appEnvironments
   }
+
+  public async run(): Promise<AppCommandOutput> {
+    throw new AbortError('You should not call this method directly, implement it in a subclass')
```

Comment:
Given the class is abstract, could the method be too? https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members:~:text=abstract%20getName()%3A%20string%3B